### ISSUE TITLE
NaN coordinates pass validate_bbox and Boxes.from_tensor validation

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4238]
Two bbox validation paths accept NaN coordinates: `validate_bbox` can return `True`, and `Boxes.from_tensor(..., validate_boxes=True)` can construct a box with NaN extents. This lets invalid coordinates pass the validation boundary and propagate into downstream geometry.

## Reproduction

```python
import torch
from kornia.geometry.bbox import validate_bbox
from kornia.geometry.boxes import Boxes

vertices = torch.tensor([
    [[0.0, 0.0], [float("nan"), 0.0], [1.0, 1.0], [0.0, 1.0]]
])
print(validate_bbox(vertices))
# True

xyxy = torch.tensor([[0.0, 0.0, float("nan"), 1.0]])
boxes = Boxes.from_tensor(xyxy, mode="xyxy", validate_boxes=True)
print(boxes.to_tensor("xyxy"))
# tensor([[nan, 0., nan, 1.]])
```

## Expected behavior

NaN coordinates should not be certified as valid. In the eager validation paths shown above, `validate_bbox` should return `False`, and `Boxes.from_tensor(..., validate_boxes=True)` should reject the input with a meaningful error. Preserve the existing opt-out/export policy rather than adding data-dependent validation to paths that intentionally bypass it.

## Cause

- [bbox.py](https://github.com/kornia/kornia/blob/3f102790/kornia/geometry/bbox.py#L80-L94) computes opposite-edge differences and rejects only when a difference is greater than `1e-4`. A NaN difference does not satisfy that comparison, so validation falls through to `True`.
- [boxes.py](https://github.com/kornia/kornia/blob/3f102790/kornia/geometry/boxes.py#L140-L145) rejects widths/heights only when they are `<= 0`. A NaN extent does not satisfy that comparison either, so construction succeeds despite `validate_boxes=True`.

The bbox/Boxes batch-4a convention audit and its current pins did not cover this case. This is distinct from the already tracked inclusive-width, vertex-order, rank, and list-padding issues.

## Suggested regression scope

Cover a NaN in each coordinate position, invalid rows mixed with valid rows, and the supported eager input layouts/modes. Check coordinate finiteness rather than relying only on derived widths: a non-finite origin must not escape because its extent was supplied separately. Include infinities in the finite-input policy tests as well; the executed reproductions above establish the NaN cases specifically. Keep valid finite inputs and the explicit validation opt-out behavior unchanged.

## Verification

Both reproductions executed on Kornia `0.9.0rc1`, commit `3f102790`, PyTorch `2.14.0`, Python `3.11.14`, macOS CPU, float32. CUDA, MPS, and export behavior were not tested for this finding. The existing convention/wart selection across conversions, bbox/boxes, and geometry transforms remains green: 343 passed, 17 xfailed, 2 skipped on CPU float32/float64.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-6-astra).
